### DISCUSSION TITLE
How to protect proprietary source code?

### DISCUSSION THREAD

**Discussion by chrisflesher** [jax-ml/jax#18999] [Q&A]
Is there a way to compile proprietary source code to protect / obfuscate it? Like if you compiled a function to XLA on one machine and then ran it on an insecure physical (not cloud) machine?

I saw there is some pickling functionality but don't think this is exactly what I'm looking for?

**Comment by jakevdp** [comment]
It sounds like what you're looking for is capability to do Ahead-of-time compilation in a way that will produce portable compilation artifacts that can be loaded on other systems. I'm not aware of any supported feature along these lines in JAX: ahead-of-time compilation tools currently focus on non-portable artifacts used for cacheing within a single system.
